File: backend/src/main/java/com/apocalypse/thefall/mapper/character/stats/PerkMapper.java
package com.apocalypse.thefall.mapper.character.stats;

import com.apocalypse.thefall.dto.character.stats.PerkDto;
import com.apocalypse.thefall.dto.character.stats.PerkInstanceDto;
import com.apocalypse.thefall.entity.character.stats.Perk;
import com.apocalypse.thefall.entity.character.stats.PerkInstance;
import org.mapstruct.Mapper;
import org.mapstruct.Mapping;

import java.util.List;

@Mapper(componentModel = "spring")
public interface PerkMapper {

    @Mapping(target = "perkId", source = "perk.id")
    PerkInstanceDto toDto(PerkInstance perkInstance);

    PerkDto toDto(Perk perk);

    List<PerkDto> toDto(List<Perk> perks);

}

File: backend/src/main/java/com/apocalypse/thefall/mapper/character/stats/perk/PerkMapper.kt
package com.apocalypse.thefall.mapper.character.stats;

import com.apocalypse.thefall.dto.character.stats.DataItemDto;
import com.apocalypse.thefall.dto.character.stats.DataTaggedItemInstanceDto;
import com.apocalypse.thefall.entity.character.stats.Perk;
import com.apocalypse.thefall.entity.character.stats.PerkInstance;
import org.mapstruct.Mapper;

import java.util.List;

@Mapper(componentModel = "spring")
public interface PerkMapper {

    DataTaggedItemInstanceDto toDto(PerkInstance perkInstance);

    DataItemDto toDto(Perk perk);

    List<DataItemDto> toDto(List<Perk> perks);

}